spring-cloud-gcp-data-spanner/src/test/java/org/springframework/cloud/gcp/data/spanner/repository/it/SpannerRepositoryIntegrationTests.java

```diff
@@ -246,6 +255,28 @@ public void queryMethodsTest() {
 		assertThat(this.subTradeRepository.count()).isEqualTo(2);
 		assertThat(this.subTradeComponentRepository.count()).isEqualTo(3);
 
+		// test eager-fetch in @Query
+		Mockito.clearInvocations(spannerTemplate);
+		final Trade aTrade = someTrade;
+		assertThat(tradeRepository.fetchByIdEagerly(aTrade.getId()))
+				.isNotEmpty()
+				.hasValueSatisfying(t -> assertThat(t.getId()).isEqualTo(aTrade.getId()))
+				.hasValueSatisfying(t -> assertThat(t.getTraderId()).isEqualTo(aTrade.getTraderId()))
+				.hasValueSatisfying(t -> assertThat(t.getSymbol()).isEqualTo(aTrade.getSymbol()))
+				.hasValueSatisfying(t -> assertThat(t.getSubTrades()).hasSize(aTrade.getSubTrades().size()));
+		Mockito.verify(spannerTemplate, Mockito.times(1))
+				.executeQuery(any(Statement.class), any(SpannerQueryOptions.class));
+
+		Mockito.clearInvocations(spannerTemplate);
+		assertThat(tradeRepository.fetchByIdLazy(aTrade.getId()))
+				.isNotEmpty()
+				.hasValueSatisfying(t -> assertThat(t.getId()).isEqualTo(aTrade.getId()))
+				.hasValueSatisfying(t -> assertThat(t.getTraderId()).isEqualTo(aTrade.getTraderId()))
+				.hasValueSatisfying(t -> assertThat(t.getSymbol()).isEqualTo(aTrade.getSymbol()))
+				.hasValueSatisfying(t -> assertThat(t.getSubTrades()).hasSize(aTrade.getSubTrades().size()));
```

Comment:
the `Trade` has default value for `subTrades` field like
```java
	@Interleaved
	private List<SubTrade> subTrades = Collections.emptyList();
```
Than is why such non-null validation in class `SpannerTemplate` line 588 is not enouph
```java
					//an interleaved property can only be List
					List propertyValue = (List) accessor.getProperty(spannerPersistentProperty);
					if (propertyValue != null) {
						resolveChildEntities(propertyValue, null);
						return;
					}
```
It was added after the rev: 26e9adf

Reply:
Good catch. How do you think that can be fixed?
Maybe we should nullify the laze loaded fields on the parent entity retrieval?

Reply:
I've tried to tell about this problem we could potentially have before but my Eng is not so fluent to be clear, sorry.
I have no clever idea yet :(
I think the most strong solution could be to have some additional flag in a`spannerPersistentProperty` that it has been populated already. 

"nulling" of fields is not a good idea - the field could be created by Lombok builder and it could be a business logic (like enty-to-json transformers) that expect the field can never be null

Reply:
Discussing with @dmitry-s offline, he pointed out that it would just be null until someone tries to access it, at which point it will be lazy-loaded. So, no one should ever see it as null.

Reply:
It sounds like a hack.
What the next step should be? Are we going to merge this PR and will do a fix later or we can collaborate somehow in another way?

Reply:
I will add a fix in this branch.

Reply:
so little I know about the GitHub : (

Reply:
Actually, it has nothing to do with the default value. The thing is, `subTrades` is not a lazy loaded property. That's why when we convert result rows to entities, we expect the field to be present. If it is not, we assume that there are no children and assign an empty list.

Reply:
Sorry, my test was wrong. You are right, it is because of the default value.

Reply:
in theory, we can to some "magic proxy" for each interleaved ... but it seems to be too complex to be stable

Reply:
we can do a very simple decision - is the code reached line 593 of SpannerTemplate line it means that any eager-interleaved is already fetched so any other should be a proxy for "lazy-interleaved"
I think the only scenario when it is not true - it is @Query(fetchInterleaved=false)
Is not it? 
Probably I'm wrong...

Reply:
in my implementation of PR #2132 I built an SQL-query (ResultSet) and a function to process it at the same time in parallel - it was a "pair". So when I'm started to process RS I had only to invoke a function.
Your PR is much simple so it is better, but there is a single place here where it is difficult to understand which fields are populated already. 

I don't understand how this magic method works (ConverterAwareMappingSpannerEntityProcessor#mapToList(com.google.cloud.spanner.ResultSet, java.lang.Class<T>, java.util.Set<java.lang.String>, boolean))
```
entityReader.read(entityClass,
	resultSet.getCurrentRowAsStruct(),
	includeColumns,
	allowMissingColumns)
```
And I can't propose any solution yet :(
But this method is a good place to mark any Interleaved as 'populated' at processing of it